sdk/search/Azure.Search.Documents/src/SearchServiceClient.cs

```diff
@@ -330,82 +593,1058 @@ public virtual SearchIndexClient GetSearchIndexClient(string indexName)
+        /// <returns>The <see cref="Response{T}"/> from the server containing the requested <see cref="SynonymMap"/>.</returns>
+        /// <exception cref="ArgumentNullException">Thrown when <paramref name="synonymMapName"/> is null.</exception>
+        /// <exception cref="RequestFailedException">Thrown when a failure is returned by the Search service.</exception>
+        [ForwardsClientCalls]
+        public virtual Response<SynonymMap> GetSynonymMap(
+            string synonymMapName,
+            SearchRequestOptions options = null,
+            CancellationToken cancellationToken = default) =>
+            SynonymMapsClient.Get(
+                synonymMapName,
+                options?.ClientRequestId,
+                cancellationToken);
+
+        /// <summary>
+        /// Gets a specific <see cref="SynonymMap"/>.
+        /// </summary>
+        /// <param name="synonymMapName">Required. The name of the <see cref="SynonymMap"/> to get.</param>
+        /// <param name="options">Optional <see cref="SearchRequestOptions"/> to customize the operation's behavior.</param>
+        /// <param name="cancellationToken">Optional <see cref="CancellationToken"/> to propagate notifications that the operation should be canceled.</param>
+        /// <returns>The <see cref="Response{T}"/> from the server containing the requested <see cref="SynonymMap"/>.</returns>
+        /// <exception cref="ArgumentNullException">Thrown when <paramref name="synonymMapName"/> is null.</exception>
+        /// <exception cref="RequestFailedException">Thrown when a failure is returned by the Search service.</exception>
+        [ForwardsClientCalls]
+        public virtual async Task<Response<SynonymMap>> GetSynonymMapAsync(
+            string synonymMapName,
+            SearchRequestOptions options = null,
+            CancellationToken cancellationToken = default) =>
+            await SynonymMapsClient.GetAsync(
+                synonymMapName,
+                options?.ClientRequestId,
+                cancellationToken)
+                .ConfigureAwait(false);
+
+        /// <summary>
+        /// Gets a list of all synonym maps.
+        /// </summary>
+        /// <param name="select">Optional property names to select. The default is all properties.</param>
+        /// <param name="options">Optional <see cref="SearchRequestOptions"/> to customize the operation's behavior.</param>
+        /// <param name="cancellationToken">Optional <see cref="CancellationToken"/> to propagate notifications that the operation should be canceled.</param>
+        /// <returns>The <see cref="Response{T}"/> from the server containing a list of <see cref="SynonymMap"/>.</returns>
+        /// <exception cref="RequestFailedException">Thrown when a failure is returned by the Search service.</exception>
+        [ForwardsClientCalls]
+        public virtual Response<IReadOnlyList<SynonymMap>> GetSynonymMaps(
```

Comment:
Should we make all these `Get*s` `Pageable<T>`?  We can discuss it after this Preview with Bruce to get an idea of how these features might grow over time.

Reply:
I opened #11053 to track, but the models themselves aren't currently pageable, and I imagine they'd never yield enough results to bother making them that way. As you said, we can discuss later.